pkgs/top-level/all-packages.nix

```diff
@@ -14377,6 +14377,10 @@ let
 
   ### SCIENCE / MATH
 
+  caffe = callPackage ../applications/science/math/caffe {
+    cudaSupport = config.caffe.cudaSupport or config.cudaSupport or true;
```

Comment:
I would have thought so too, maybe I misunderstand how these expressions are evaluated but when I explicitly set config.cudaSupport to false it actually sets cudaSupport to false here.